components/infra-proxy-service/server/roles.go

```diff
@@ -75,23 +84,11 @@ func (s *Server) GetRole(ctx context.Context, req *request.Role) (*response.Role
 		return nil, status.Errorf(codes.InvalidArgument, "invalid org ID: %s", err.Error())
 	}
 
-	result := RoleListResult{}
-	newReq, err := c.client.NewRequest("GET", "search/role", nil)
+	result, err := c.GetRoleList()
 	if err != nil {
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	res1, err := c.client.Do(newReq, &result)
```

Comment:
DRY the code and added `GetRoleList` function to fetch roles from Chef Infra Server search API.